### DISCUSSION TITLE
SPWG - Data legality discussion

### DISCUSSION THREAD

**Discussion by stuberman** [filecoin-project/community#221] [Mining]
This discussion is around the perceived problems with storing data that may be illegal or controlled. Please add to the list of concerns, provide a use case, and optionally how to mitigate or resolve.

1. If law enforcement or legal representatives demand that a piece CID (specific data within a sector) is deleted, how would a storage provider delete that data? Today we might be forced to delete a whole sector with other valid deals.

2. If law enforcement or legal representatives demand that some data is deleted, but they don't have a CID, then how would we comply?

3.  How do we transfer some liability to clients who want to store data?

**Comment by JDLH** [comment]
Thank you for creating this topic. Here is a use case which concerns me: Suppose I am a storage miner. Someone offers me a deal to store their data for a fee. Their data turns out to be illegal in some way, such that mere possession of the data opens me to prosecution or punishment. (Example: copyright-infringing content, or child exploitation content.) What can I as a storage miner do to protect myself from this risk.  

Variant use case: the content is encrypted such that I have no idea that it is illegal, but the prosecuting authorities can decrypt the data and prove that it is illegal. What can I as a storage miner do to protect myself from the risk of possessing data which I have no way of knowing is illegal?
